java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -3069,88 +3079,188 @@ void testWindowingLead() {
+                 new StructData(-111, "s111"), new StructData(null, "s112"), new StructData(-222, "s222"), new StructData(-333, "s333"));
+
+             Table windowAggResults = sorted.groupBy(0, 1)
+                 .aggregateWindows(Aggregation
+                     .lead(1, defaultOutput)
+                     .onColumn(3)
+                     .overWindow(options));
+             Table decWindowAggResults = sorted.groupBy(0, 4)
+                 .aggregateWindows(Aggregation
+                     .lead(1, decDefaultOutput)
+                     .onColumn(6)
+                     .overWindow(options));
+             Table listWindowAggResults = sorted.groupBy(0, 1).aggregateWindows(
+                 Aggregation
+                     .lead(1, listDefaultOutput)
+                     .onColumn(7)
+                     .overWindow(options));
+             Table structWindowAggResults = sorted.groupBy(0, 1).aggregateWindows(
+                 Aggregation
+                     .lead(1, structDefaultOutput)
+                     .onColumn(8)
+                     .overWindow(options));
+             ColumnVector expectAggResult = ColumnVector.fromBoxedInts(5, 1, 9, -3, 9, 8, 2, -7, 0, 6, 6, -11);
+             ColumnVector decExpectAggResult = ColumnVector.decimalFromLongs(-2, 5, 1, 9, -3, 9, 8, 2, -7, 0, 6, 6, -11);
+             ColumnVector listExpectAggResult = ColumnVector.fromLists(
+                 new HostColumnVector.ListType(true, new HostColumnVector.BasicType(true, DType.INT32)),
+                 Arrays.asList(14, null, 15, null), Arrays.asList((Integer) null), Arrays.asList(16), Arrays.asList(444, null, 555),
+                 Arrays.asList(23, 24), Arrays.asList(25, 26, 27), Arrays.asList(28, 29, null), Arrays.asList(-44),
+                 Arrays.asList(32, 33, 34), Arrays.asList(35, 36), Arrays.asList(37, 38, 39), Arrays.asList(6, null, null));
+             ColumnVector structExpectAggResult = ColumnVector.fromStructs(
+                 new StructType(true,
+                     new BasicType(true, DType.INT32),
+                     new BasicType(true, DType.STRING)),
+                 new StructData(null, "s2"), new StructData(2, null), new StructData(3, "s3"), new StructData(-3, "s3"),
+                 null, new StructData(13, "s13"), new StructData(14, "s14"), new StructData(-14, "s14"),
+                 new StructData(null, "s112"), new StructData(2, "s222"), new StructData(3, "s333"), new StructData(-333, "s333"))) {
+          assertColumnsAreEqual(expectAggResult, windowAggResults.getColumn(0));
+          assertColumnsAreEqual(decExpectAggResult, decWindowAggResults.getColumn(0));
+          assertColumnsAreEqual(listExpectAggResult, listWindowAggResults.getColumn(0));
+
+          // TODO  this is not gonna work, since cudf has some issue on lead with default values
+          // assertColumnsAreEqual(structExpectAggResult, structWindowAggResults.getColumn(0));
```

Comment:
Please raise a `libcudf` bug for this. It looks like I neglected to add a test for `struct` lead/lag with defaults, so my tests didn't catch this.

Reply:
Actually, I wonder if this might be resolved, thanks to @sperlingxx's fix in https://github.com/rapidsai/cudf/pull/8211.

@wbo4958, is this still failing with @sperlingxx's fix? If yes, no worries. We'll hunt it down in the next release. :]

Reply:
Hi @mythrocks Filed a issue for you https://github.com/rapidsai/cudf/issues/8281

Reply:
Yes, it's still failing even with @sperlingxx's fix